course3/homework-unique_ptr/unique_ptr.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <cstddef>
+
+
+namespace karun
+{
+	template <class T>
+	class unique_ptr
+	{	
+		
+	public:
+		unique_ptr();
+		unique_ptr(std::nullptr_t);
```

Comment:
Ah okay. I thought it prevented all implicit conversions, which doesn't make sense in retrospect. It's clear now, thanks!